### ISSUE TITLE
AARCH64 Support

### ISSUE THREAD

**Issue by nikstur** [randombit/botan-rs#114]
Does botan-rs support aarch64? I'm trying to compile it without vendoring but it complains about a bunch of mismatched types:

```sh
error[E0308]: mismatched types
   --> /build/cargo-vendor-dir/botan-0.10.6/src/utils.rs:146:31
    |
146 |     let cstr = CStr::from_ptr(raw_cstr);
    |                -------------- ^^^^^^^^ expected `*const u8`, found `*const i8`
    |                |
    |                arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`
```

**Comment by thillux** [comment]
```diff
diff --git a/botan/src/utils.rs b/botan/src/utils.rs
index 674897c..1a6d781 100644
--- a/botan/src/utils.rs
+++ b/botan/src/utils.rs
@@ -142,7 +142,7 @@ fn cstr_slice_to_str(raw_cstr: &[u8]) -> Result<String> {
 }

 #[cfg(feature = "botan3")]
-unsafe fn cstr_to_str(raw_cstr: *const i8) -> Result<String> {
+unsafe fn cstr_to_str(raw_cstr: *const u8) -> Result<String> {
     let cstr = CStr::from_ptr(raw_cstr);
     Ok(cstr.to_str().map_err(Error::conversion_error)?.to_owned())
 }
```
Seems to fix this.

**Comment by randombit** [comment]
Ah yes that type should definitely be `c_char` - thanks!

**Comment by randombit** [comment]
Fixed now in `botan 0.10.7`
